subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileIntegrationTest.groovy

```diff
@@ -476,4 +477,80 @@ class JavaCompileIntegrationTest extends AbstractIntegrationSpec {
         then:
         skipped ':compileJava'
     }
+
+    @Issue("gradle/gradle#1358")
+    def "compile classpath snapshotting should warn when jar on classpath is malformed"() {
+        buildFile << '''
+            apply plugin: 'java'
+            
+            dependencies {
+               compile files('foo.jar')
+            }
+        '''
+        file('foo.jar') << 'this is clearly not a well formed jar file'
+        file('src/main/java/Hello.java') << 'public class Hello {}'
+
+        when:
+        executer.withFullDeprecationStackTraceDisabled()
+        executer.expectDeprecationWarning()
+        run 'compileJava'
+
+        then:
+        executedAndNotSkipped ':compileJava'
+
+    }
+
+    @Issue("gradle/gradle#1358")
+    def "compile classpath snapshotting should warn when jar on classpath contains malformed class file"() {
+        buildFile << '''
+            apply plugin: 'java'
+            
+            task fooJar(type:Jar) {
+                archiveName = 'foo.jar'
+                from file('foo.class')
+            }
+            
+            dependencies {
+               compile files(fooJar.archivePath)
+            }
+            
+            compileJava.dependsOn(fooJar)
+            
+            
+        '''
+        file('foo.class') << 'this is clearly not a well formed class file'
+        file('src/main/java/Hello.java') << 'public class Hello {}'
+
+        when:
+        executer.withFullDeprecationStackTraceDisabled()
+        executer.expectDeprecationWarning()
+        run 'compileJava'
+
+        then:
+        executedAndNotSkipped ':fooJar', ':compileJava'
+    }
+
+    @Issue("gradle/gradle#1358")
+    def "compile classpath snapshotting should warn when class on classpath is malformed"() {
+        buildFile << '''
+            apply plugin: 'java'
+            
+            dependencies {
+               compile files('classes')
+            }
+            
+        '''
+        file('classes/foo.class') << 'this is clearly not a well formed class file'
+        file('src/main/java/Hello.java') << 'public class Hello {}'
+
+        when:
+        executer.withFullDeprecationStackTraceDisabled()
+        executer.expectDeprecationWarning()
+        run 'compileJava'
+
+        then:
+        executedAndNotSkipped ':compileJava'
+    }
+
```

Comment:
You're right, this will only happen if we fall back to scanning the compile classpath and in that case the compile classpath snapshotter will already warn. So that speaks for a low log level and no deprecation warning.